File: frontend/src/types/brand.ts
export interface Brand {
  id: string;
  name: string;
  slug: string;
  description: string | null;
  website: string | null;
  logo_url: string | null;
  active: boolean;
  created_at: string;
  updated_at: string;
}

export interface BrandCreate {
  name: string;
  slug: string;
  description?: string | null;
  website?: string | null;
  logo_url?: string | null;
  active?: boolean;
}

export interface BrandUpdate {
  name?: string;
  slug?: string;
  description?: string | null;
  website?: string | null;
  logo_url?: string | null;
  active?: boolean;
}

File: backend/app/schemas/brand.py
from datetime import datetime

from pydantic import BaseModel


class BrandCreate(BaseModel):
    name: str
    slug: str
    description: str | None = None
    website: str | None = None
    logo_url: str | None = None
    active: bool = True


class BrandUpdate(BaseModel):
    name: str | None = None
    slug: str | None = None
    description: str | None = None
    website: str | None = None
    logo_url: str | None = None
    active: bool | None = None


class BrandRead(BaseModel):
    id: str
    name: str
    slug: str
    description: str | None
    website: str | None
    logo_url: str | None
    active: bool
    created_at: datetime
    updated_at: datetime

    model_config = {"from_attributes": True}
